packages/workflow/src/ErrorReporterProxy.ts

```diff
@@ -0,0 +1,28 @@
+export interface IReportingOptions {
+	tags?: Record<string, string>;
+	extra?: Record<string, unknown>;
+}
+
+interface ErrorReporter {
+	error: (error: Error, options?: IReportingOptions) => void;
+	warn: (warning: Error | string, options?: IReportingOptions) => void;
+}
+
+const isProduction = process.env.NODE_ENV === 'production';
+
+const instance: ErrorReporter = {
+	error: (error) => isProduction && console.error('ERROR', error.message, error.stack),
+	warn: (warning) => isProduction && console.warn('WARN', warning),
+};
+
+export function init(errorReporter: ErrorReporter) {
+	instance.error = errorReporter.error;
+	instance.warn = errorReporter.warn;
+}
+
+export const error = (e: unknown, options?: IReportingOptions) => {
+	if (e instanceof Error) instance.error(e, options);
```

Comment:
If a lib throws e.g. a string literal this won't report it? Maybe `error` and `warn` should have identical signatures.

Reply:
made `error` and `warn` consistent.